modules/core/arrow-test/src/main/kotlin/arrow/test/laws/UnalignLaws.kt

```diff
@@ -0,0 +1,90 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.Ior
+import arrow.core.Tuple2
+import arrow.core.extensions.eq
+import arrow.core.extensions.ior.eq.eq
+import arrow.core.extensions.tuple2.eq.eq
+import arrow.core.toT
+import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Semialign
+import arrow.typeclasses.Unalign
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object UnalignLaws {
+  fun <F> laws(
+    UA: Unalign<F>,
+    gen: Gen<Kind<F, Int>>,
+    EQK: EqK<F>
+  ): List<Law> = SemialignLaws.laws(UA, gen, EQK) + unalignLaws(UA, gen, EQK)
+
+  fun <F> laws(
+    UA: Unalign<F>,
+    gen: Gen<Kind<F, Int>>,
+    EQK: EqK<F>,
+    FOLD: Foldable<F>
+  ): List<Law> = SemialignLaws.laws(UA, gen, EQK, FOLD) + unalignLaws(UA, gen, EQK)
+
+  private fun <F> unalignLaws(
+    UA: Unalign<F>,
+    gen: Gen<Kind<F, Int>>,
+    EQK: EqK<F>
+  ): List<Law> {
+    val iorIntEq = buildEq(EQK, Ior.eq(Int.eq(), Int.eq()))
+    val intEq = buildEq(EQK, Int.eq())
+    val tuple2Eq = Tuple2.eq(intEq, intEq)
+
+    return listOf(
+      Law("Unalign Laws: split union shape into its components #1") { UA.law1(gen, tuple2Eq) },
```

Comment:
Names! These laws are testing roundtrips: `align . unalign == id` an `unalign . align == id` (Ignoring currying). I suck at naming things tho, but there must be something better :)

Reply:
ok, hope you like the new names better

Reply:
Oh I suck at naming things, otherwise I would have suggested something, but numbers aren't that great :) Thanks